### PR TITLE
chore(deps): Upgrade Sentry Python SDK to 2.16.0

### PR DESCRIPTION
**Pull request by BYK**
This version comes with Brotli compression and HTTP2Transport which are enabled as part of this patch.

We already have `httpcore>=1.*` and `Brotli` as part of our dependencies so no changes needed in this patch regarding getting these installed.

### CODE REVIEW

**Review by sentrivana — APPROVED** [review]

File: src/sentry/utils/sdk.py [hunk 1]
```diff
@@ -280,6 +280,9 @@ def _get_sdk_options() -> tuple[SdkConfig, Dsns]:
     sdk_options["release"] = (
         f"backend@{sdk_options['release']}" if "release" in sdk_options else None
     )
+    sdk_options.setdefault("_experiments", {}).update(
+        transport_http2=True,
+    )
```
Comment (asottile-sentry) [inline src/sentry/utils/sdk.py:285]: just to minimize risk a bit -- should the upgrade and enabling this be two separate patches? (would also allow fast reverting one without the other if it goes poorly)

Comment (BYK) [reply src/sentry/utils/sdk.py:285]: Yeah, makes sense. I was actually hoping to trigger this with a feature flag but not sure if _that_ makes sense.

Will update this to not have http2 turned on.

Comment (BYK) [reply src/sentry/utils/sdk.py]: @asottile-sentry reverted this part.

**Review by asottile-sentry — APPROVED** [review]

**Comment by asottile-sentry** [comment]
reverting as this broke the development environment:

```
20:21:49 monitors-clock-tick                   |   File "/Users/lyn/Code/sentry/.venv/lib/python3.12/site-packages/sentry_sdk/client.py", line 259, in __init__
20:21:49 monitors-clock-tick                   |     self._init_impl()
20:21:49 monitors-clock-tick                   |   File "/Users/lyn/Code/sentry/.venv/lib/python3.12/site-packages/sentry_sdk/client.py", line 402, in _init_impl
20:21:49 monitors-clock-tick                   |     self.spotlight = setup_spotlight(self.options)
20:21:49 monitors-clock-tick                   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
20:21:49 monitors-clock-tick                   |   File "/Users/lyn/Code/sentry/.venv/lib/python3.12/site-packages/sentry_sdk/spotlight.py", line 118, in setup_spotlight
20:21:49 monitors-clock-tick                   |     settings.MIDDLEWARE.append("sentry_sdk.spotlight.SpotlightMiddleware")
20:21:49 monitors-clock-tick                   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
20:21:49 monitors-clock-tick                   | AttributeError: 'tuple' object has no attribute 'append'
```
